test/Uri.js

```diff
@@ -48,8 +48,39 @@ describe('Uri', function() {
 		assert.strictEqual('', uri.getOrigin());
 	});
 
+	it('should set protocol http: on localhost uri', function() {
+		var uri = new Uri('localhost:8080');
+		assert.strictEqual('http:', uri.getProtocol());
+		assert.strictEqual('8080', uri.getPort());
+	});
+
+	it('should support useful types of uri schemes like "tel"', function() {
+		var uri = new Uri('tel:hostname');
+		assert.strictEqual('tel:', uri.getProtocol());
+	});
+
+	it('should support uri schemes that include hyphens', function() {
+		var uri = new Uri('ms-excel:hostname');
+		assert.strictEqual('ms-excel:', uri.getProtocol());
+	});
+
+	it('should support uri schemes that include numerical values', function() {
+		var uri = new Uri('pkcs11:hostname');
+		assert.strictEqual('pkcs11:', uri.getProtocol());
+	});
+
+	it('should support uri schemes that include periods', function() {
+		var uri = new Uri('iris.beep:hostname');
+		assert.strictEqual('iris.beep:', uri.getProtocol());
+	});
+
+	it('should support uri schemes that include pluses', function() {
+		var uri = new Uri('some+scheme:hostname');
+		assert.strictEqual('some+scheme:', uri.getProtocol());
+	});
+
 	it('should support set port on uri', function() {
-		var uri = new Uri('hostname:8080');
+		var uri = new Uri('http://hostname:8080');
```

Comment:
Hey, @jbalsas. I also agree with Eduardo. I was trying to figure out if there was any way to make this working without adding a parameter, but I think it's not the case here.

I would like to hear from you if worth to add this new support. I think it worth. Many users will try to use the component this way like @kienD did, and there is no clear mention about the restriction.

Like you've suggested, we can add a new parameter that skips `maybeAddProtocolAndHostname_`.